### PR TITLE
Straight

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

File: init.el [hunk 1]
```diff
@@ -147,16 +164,16 @@
 (require 'config-looks)
 
 (use-package dash
-  :ensure
+  :straight t
```
Comment (drrlvn) [inline init.el]: Is the `t` needed? Can't we just use `:straight` like we did with `:ensure`?

File: init.el [hunk 2]
```diff
@@ -17,14 +17,30 @@
...
+      (goto-char (point-max))
+      (eval-print-last-sexp)))
+  (load bootstrap-file nil 'nomessage))
+
+(require 'straight)
+(setq straight-profiles
+      `((nil . ,(concat user-emacs-directory "lockfile.el"))
+        (site . ,(concat my/site-config-directory "lockfile.el"))))
+
+(require 'seq)
```
Comment (drrlvn) [inline init.el]: Why is this needed?

File: init.el [hunk 3]
```diff
@@ -17,14 +17,30 @@
       inhibit-message t
       load-prefer-newer t
       custom-file (concat user-emacs-directory "custom.el")
-      package-user-dir (concat user-emacs-directory "elpa")
```
Comment (drrlvn) [inline init.el]: Shouldn't we leave this in case the user uses `package` for something?

**Comment by drrlvn** [comment]
The CI should also be fixed and should ignore the lockfile.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -15,4 +15,4 @@ jobs:
         sudo apt update
         sudo apt install emacs26
     - name: Byte compile files
-      run: git ls-files '*.el' ':!:.*' -z | xargs -0 emacs -Q --batch -l .ci.el -f 'batch-byte-compile'
+      run: git ls-files '*.el' ':!:.*' --exclude=lockfile.el -z | xargs -0 emacs -Q --batch -l .ci.el -f 'batch-byte-compile'
```
Comment (drrlvn) [inline .github/workflows/ci.yml]: This doesn't work (at least not how you think it does), that's why I used `':!:.*'` and not `--exclude`.

**Comment by r-darwish** [comment]
@vmalloc note that once this is merged, the next Emacs startup could take ~20 minutes, so do that in your coffee time 🙂

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -15,4 +15,4 @@ jobs:
         sudo apt update
         sudo apt install emacs26
     - name: Byte compile files
-      run: git ls-files '*.el' ':!:.*' -z | xargs -0 emacs -Q --batch -l .ci.el -f 'batch-byte-compile'
+      run: git ls-files '*.el' ':!:.*' ':!:lockfile.el' -z | xargs -0 emacs -Q --batch -l .ci.el -f 'batch-byte-compile'&
```
Comment (drrlvn) [inline .github/workflows/ci.yml]: Is the `&` there to make this run in the background? It doesn't seem to work.

File: lisp/config-custom.el [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ;;; config-custom.el --- custom configuration -*- lexical-binding: t; byte-compile-warnings: (not unresolved) -*-
 ;;; Commentary:
 ;;; Code:
+(require 'seq)
```
Comment (drrlvn) [inline lisp/config-custom.el]: You moved this but didn't answer - why is this needed?

File: init.el [hunk 7]
```diff
@@ -1029,29 +1027,33 @@ _M-p_: Unmark  _M-n_: Unmark  _q_: Quit"
...
 (use-package savehist
   :config (savehist-mode 1))
 
 (use-package kubernetes
-  :ensure
+  :straight t
   :bind (("C-c k" . kubernetes-overview)))
 
-(my/load-if-exists (concat my/site-config-directory "config.el"))
+(use-package git-timemachine
```
Comment (drrlvn) [inline init.el]: This shouldn't be in this PR.

File: lisp/straight-bootstrap.el [hunk 8]
```diff
@@ -0,0 +1,15 @@
+(defvar bootstrap-version)
```
Comment (drrlvn) [inline lisp/straight-bootstrap.el]: I think you need to add the documentation emacs requires for files in the top and bottom (see other `.el` files we have).

File: init.el [hunk 9]
```diff
@@ -804,14 +804,15 @@
...
-    :config
-    (push '("git.infinidat.com" "git.infinidat.com/api/v4" "git.infinidat.com" forge-gitlab-repository) forge-alist)))
+(use-package forge
+  :straight t
+  :if (eq system-type 'windows-nt)
+  :after magit
+  :config
+  (push '("git.infinidat.com" "git.infinidat.com/api/v4" "git.infinidat.com" forge-gitlab-repository) forge-alist))
+
+(eval-when-compile
```
Comment (drrlvn) [inline init.el]: Why was this moved after the package?

File: init.el (hunk 1 above)
Comment (drrlvn) [reply init.el]: Can you open a ticket then, please?

**Comment by r-darwish** [comment]
I think I finally managed to solve every issue. We're good to merge (I think)

File: init.el [hunk 10]
```diff
@@ -813,25 +811,27 @@
...
-    :config
-    (push '("git.infinidat.com" "git.infinidat.com/api/v4" "git.infinidat.com" forge-gitlab-repository) forge-alist)))
+(use-package forge
+  :straight t
+  :init (when (eq system-type 'windows-nt)
+          (setq forge-bug-reference-hooks ()))
+  :after magit
+  :config
+  (push '("git.infinidat.com" "git.infinidat.com/api/v4" "git.infinidat.com" forge-gitlab-repository) forge-alist)
+  (when (eq system-type 'windows-nt)
```
Comment (drrlvn) [inline init.el]: Maybe add a comment explaining this is a workaround with a link to the issue?

**Comment by r-darwish** [comment]
Finally 😄 

https://www.youtube.com/watch?v=pOTwvh4B3jM

**Comment by r-darwish** [comment]
@vmalloc @idanski @gleon99 - your next Emacs startup after pulling this might take something like ~20 minutes, so I suggest that you either do it on your idle time, or keep a working copy of Emacs running during this process so you can continue your work. After this process is completed you can delete `~/.emacs.d/elpa`
